### PR TITLE
Adding materialUp API

### PR DESCRIPTION
**Pull request by Cutta**
http://www.materialup.com/ is also provides design news & inspiration and have an API.
Integrating https://material.uplabs.com/users/api
API includes these services:
**/v1/posts?days_ago** = Get Posts Daily
**/v1/posts/post_id** = Get Post Detail
**/v1/posts/post_id/comments** = Get Post's Comment
I have integrated them  inspired by Dribbble API integration.

screenshot_2016-10-05-16-40-35
screenshot_2016-10-06-19-06-37

### CODE REVIEW

**Comment by matthieua** [comment]
👍

**Comment by nickbutcher** [comment]
Wow, thanks for the awesome PR! Sorry for the slow reply but i've been traveling. Let me take a look and respond with proper review comments soon.

**Comment by Cutta** [comment]
Hı @nickbutcher. Have you had a chance to look this PR?

File: app/src/main/java/io/plaidapp/data/api/materialup/model/Comment.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+     * 
+     * @param createdAt
+     *     The created_at
+     */
+    public void setCreatedAt(String createdAt) {
+        this.createdAt = createdAt;
+    }
+
+
+    public Spanned getParsedBody(TextView textView) {
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/data/api/materialup/model/Comment.java]: This method shouldn't be in Model-type class. If I were you, I would reuse existing `parseDribbbleHtml(String input, ColorStateList linkTextColor, @ColorInt int linkHighlightColor)` in direct class instead of creating new one, which brings no additional value.

File: app/src/main/java/io/plaidapp/data/api/materialup/model/Post.java [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+            return new Post(source);
+        }
+
+        @Override
+        public Post[] newArray(int size) {
+            return new Post[size];
+        }
+    };
+
+    public Spanned getParsedDescription(ColorStateList linkTextColor,
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/data/api/materialup/model/Post.java]: This Model class should only read and write data without modifying received data. Another thing: I don't see any additional value for this method which existing `getParsedDescription()` lacks.

File: app/src/main/java/io/plaidapp/data/prefs/MaterialUpPrefs.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+
+
+    public boolean isLoggedIn(){
+        return  true;//// TODO: 28/09/16 materialUp api does not contain login,logout
+    }
+
+    public boolean userCanPost(){
+        return true;//// TODO: 28/09/16 materialUp api does not contain login,logout
+    }
```
Comment (piotrek1543) [inline app/src/main/java/io/plaidapp/data/prefs/MaterialUpPrefs.java]: Could you fix these `TODO`'s or remove them and leave as one comment?

File: gradle.properties [hunk 4]
```diff
@@ -13,7 +13,7 @@
 # org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
 
 # When configured, Gradle will run in incubating parallel mode.
-# This option should only be used with decoupled projects. More details, visit
+# This option should only be used with decoupled projects. More degit tails, visit
```
Comment (piotrek1543) [inline gradle.properties]: Revert this change to default value.

File: build.gradle [hunk 5]
```diff
@@ -21,7 +21,7 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.3'
+        classpath 'com.android.tools.build:gradle:2.2.0'
```
Comment (piotrek1543) [inline build.gradle]: Update to ` classpath 'com.android.tools.build:gradle:2.3.0'`.

**Comment by piotrek1543** [comment]
@Cutta hi, your Pull Request is getting old and unresolvable.

 Could check my comments, fix at least a half ;-) and merge this one with new commits from 'master' to fix 'Conflicting files'.

**Comment by piotrek1543** [comment]
@nickbutcher are you gonna to put this in new release? If yes, I can fix merge conflicts and clean the code.

**Comment by nickbutcher** [comment]
Thank you for the PR but this is not something we'd like to merge in the project. In the future please feel free to open an issue prior to developing a feature to discuss.
